Problem with Metrics 4 and Datastax Java Driver for Apache Cassandra

See https://docs.datastax.com/en/developer/java-driver/3.5/manual/metrics/#metrics-4-compatibility
```
Uncaught error from thread [oms-delivery-impl-application-lagom.persistence.dispatcher-27]: com/codahale/metrics/JmxReporter, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[oms-delivery-impl-application]
java.lang.NoClassDefFoundError: com/codahale/metrics/JmxReporter
	at com.datastax.driver.core.Metrics.<init>(Metrics.java:146)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1501)
	at com.datastax.driver.core.Cluster.init(Cluster.java:208)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
	at akka.persistence.cassandra.ConfigSessionProvider.$anonfun$connect$1(ConfigSessionProvider.scala:48)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:304)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:81)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.JmxReporter
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 18 more
```

--- a/java/build.gradle.kts
+++ b/java/build.gradle.kts
@@ -15,7 +15,7 @@ object Versions {
     const val ktlint = "0.30.0"
     const val `kotlin-logging` = "1.6.25"
     const val config4k = "0.4.0"
-    const val metrics = "4.0.3"
+    const val metrics = "3.2.6"
     const val hikari = "2.7.9"
     const val kotlintest = "3.1.10"
     const val `mockito-kotlin` = "2.1.0"

--- a/java/src/main/kotlin/org/taymyr/lagom/metrics/Metrics.kt
+++ b/java/src/main/kotlin/org/taymyr/lagom/metrics/Metrics.kt
@@ -1,6 +1,7 @@
 package org.taymyr.lagom.metrics
 
 import akka.stream.Materializer
+import com.codahale.metrics.JvmAttributeGaugeSet
 import com.codahale.metrics.Meter
 import com.codahale.metrics.MetricFilter
 import com.codahale.metrics.MetricRegistry
@@ -10,9 +11,9 @@ import com.codahale.metrics.graphite.GraphiteReporter
 import com.codahale.metrics.graphite.GraphiteUDP
 import com.codahale.metrics.graphite.PickledGraphite
 import com.codahale.metrics.jvm.GarbageCollectorMetricSet
-import com.codahale.metrics.jvm.JvmAttributeGaugeSet
 import com.codahale.metrics.jvm.MemoryUsageGaugeSet
 import com.codahale.metrics.jvm.ThreadStatesGaugeSet
+import com.google.inject.ConfigurationException
 import com.google.inject.Injector
 import com.lightbend.lagom.internal.server.status.MetricsServiceImpl
 import com.lightbend.lagom.javadsl.server.status.CircuitBreakerStatus
@@ -97,8 +98,13 @@ constructor(conf: Config, @Suppress("MemberVisibilityCanBePrivate") val registry
                         }
                     }
                 }
-            } catch (e: NoClassDefFoundError) {
-                logger.error { "Libraries 'play-jdbc-api' and 'HikariCP' not found in runtime classpath for `enableHikari = true`" }
+            } catch (e: Exception) {
+                when (e) {
+                    is NoClassDefFoundError, is ConfigurationException -> {
+                        logger.error { "Libraries 'play-jdbc-api' and 'HikariCP' not found in runtime classpath for `enableHikari = true`" }
+                    }
+                    else -> logger.error(e) { "Can't enable metrics for Hikari connection pool: $e" }
+                }
             }
         }
     }

--- a/java/src/test/kotlin/org/taymyr/lagom/metrics/MetricsTest.kt
+++ b/java/src/test/kotlin/org/taymyr/lagom/metrics/MetricsTest.kt
@@ -4,10 +4,10 @@ import akka.NotUsed
 import akka.stream.javadsl.Source
 import akka.stream.javadsl.Source.from
 import akka.stream.javadsl.Source.maybe
+import com.codahale.metrics.JvmAttributeGaugeSet
 import com.codahale.metrics.MetricFilter.ALL
 import com.codahale.metrics.MetricRegistry
 import com.codahale.metrics.jvm.GarbageCollectorMetricSet
-import com.codahale.metrics.jvm.JvmAttributeGaugeSet
 import com.codahale.metrics.jvm.MemoryUsageGaugeSet
 import com.codahale.metrics.jvm.ThreadStatesGaugeSet
 import com.lightbend.lagom.internal.server.status.MetricsServiceImpl

